src/hybrid/parse_hybrid.c

```diff
@@ -466,6 +474,60 @@ static void copyRequestConfig(RequestConfig *dest, const RequestConfig *src) {
   dest->BM25STD_TanhFactor = src->BM25STD_TanhFactor;
 }
 
+// Helper function to get LIMIT value from parsed aggregation pipeline
+static size_t getLimitFromPipeline(Pipeline *pipeline) {
+  RS_ASSERT(pipeline);
+
+  PLN_ArrangeStep *arrangeStep = AGPLN_GetArrangeStep(&pipeline->ap);
+  if (arrangeStep && arrangeStep->isLimited && arrangeStep->limit > 0) {
+    return (size_t)arrangeStep->limit;
+  }
+  return 0;
+}
+
+// Helper function to check if LIMIT was explicitly provided
+static bool hasExplicitLimitInPipeline(Pipeline *pipeline) {
+  if (!pipeline) return false;
+
+  PLN_ArrangeStep *arrangeStep = AGPLN_GetArrangeStep(&pipeline->ap);
+  return (arrangeStep && arrangeStep->isLimited);
```

Comment:
Notice that a `KNN` query sets this as well. If that's still fine for the check you want to check here, let's add that to the comment